tutorials/klusterai-api/effective-prompt-engineering.ipynb

```diff
@@ -0,0 +1,731 @@
+    "\n",
+    "1. Calculate total monthly revenue for each product.\n",
+    "2. Identify the best and worst-performing products by revenue.\n",
+    "3. Suggest specific and actionable strategies to boost overall sales, particularly focusing on lower-performing products.\n",
+    "4. Provide recommendations on whether adjusting pricing or marketing might improve profitability based on current sales performance.\n",
+    "\n",
+    "Structure your response clearly with separate sections and explain briefly your reasoning behind each recommendation.\"\"\"\n",
+    "\n",
+    "# Display the prompt for visual reference\n",
+    "display(Markdown(better_prompt_text))\n",
+    "\n",
+    "# Use the variable in the API call\n",
+    "response_better = client.chat.completions.create(\n",
+    "    model=model,\n",
+    "    messages=[{\"role\": \"user\", \"content\": better_prompt_text}],\n",
+    "    temperature=0.5\n",
+    ").choices[0].message.content\n",
+    "print(\"\\nModel response:\\n\" + response_better)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f3ca7a4",
+   "metadata": {},
+   "source": [
+    "### Analysis\n",
+    "This Better prompt elevates the request beyond the \"Bad\" version by incorporating pricing details and clearer instructions, which yield more focused calculations and recommendations. The prompt also specifies a timeline (January 2025) and outlines key tasks (calculating revenue, identifying product performance, suggesting sales-boosting strategies). As a result, the LLM response provides:\n",
+    "\n",
+    "1. **Structured Revenue Calculations**: By explicitly asking for per-product revenue, it brings a quantitative foundation to the analysis.\n",
+    "2. **Actionable Recommendations**: Requests for pricing and marketing adjustments prompt specific suggestions, rather than broad, generic ideas.\n",
+    "3. **Organized Output**: Clearly defined sections ensure the response is easy to interpret and act upon.\n",
+    "\n",
+    "However, the prompt stops short of requesting profit margins, customer demographics, or broader business context (e.g., marketing channels or inventory challenges) seen in the Best prompt. Consequently, while the \"Better\" prompt moves toward more concrete, data-driven insights, it still lacks the richness and depth needed to produce thorough, strategic recommendations."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a8ceea92",
+   "metadata": {},
+   "source": [
+    "## Best prompt 🟢\n",
+    "\n",
+    "Finally, we arrive at our best prompt. This prompt represents the gold standard for eliciting high-value insights from AI models by providing comprehensive business context, detailed product information, and structured analytical requirements. By incorporating demographic data, profit margins, marketing channels, and specific business challenges, this prompt enables the model to deliver  tailored and actionable recommendations:"
```

Comment:
I believe saying the "gold standard" is a bit much haha . I would say that this prompt provides all the details necessary to achieve our ultimate goal of XXXX (the goal  should be clearly defined at the beginning and reference across all the different sections etc